typescript/lib/networks/network_context.ts

```diff
@@ -0,0 +1,440 @@
+                }
+            })
+            //TODO TODO TODO
+            // srcFp, err := os.Open(srcAbsFilepath)
+            // if err != nil {
+            //     return stacktrace.Propagate(err, "An error occurred opening static file '%v' source file '%v' for reading", staticFileId, srcAbsFilepath)
+            // }
+            // defer srcFp.Close()
+
+            // destFp, err := os.Create(destAbsFilepath)
+            // if err != nil {
+            //     return stacktrace.Propagate(err, "An error occurred opening static file '%v' destination file '%v' for writing", staticFileId, destAbsFilepath)
+            // }
+            // defer destFp.Close()
+
+            // if _, err := io.Copy(destFp, srcFp); err != nil {
+            //     return stacktrace.Propagate(err, "An error occurred copying all the bytes from static file '%v' source filepath '%v' to destination filepath '%v'", staticFileId, srcAbsFilepath, destAbsFilepath)
+            // }
+
+        }
+        return null;
+    }
+
+    // Docs available at https://docs.kurtosistech.com/kurtosis-libs/lib-documentation
+    public registerFilesArtifacts(filesArtifactUrls: Map<FilesArtifactID, string>): Error {
+        const filesArtifactIdStrsToUrls: Map<string, string> = new Map();
+        for (let artifactId in filesArtifactUrls) {
+            const url: string = filesArtifactUrls[artifactId];
+            filesArtifactIdStrsToUrls[String(artifactId)] = url;
+        }
+        const args: RegisterFilesArtifactsArgs = newRegisterFilesArtifactsArgs(filesArtifactIdStrsToUrls);
+        // TODO TODO TODO - CALLBACK & ERROR-HANDLING
+        // if _, err := networkCtx.client.RegisterFilesArtifacts(context.Background(), args); err != nil {
+        //     return stacktrace.Propagate(err, "An error occurred registering files artifacts: %+v", filesArtifactUrls)
+        // }
+        return null;
+    }
+
+    // Docs available at https://docs.kurtosistech.com/kurtosis-libs/lib-documentation
+    public addService(
+        serviceId: ServiceID,
+        containerCreationConfig: ContainerCreationConfig, //TODO
+        generateRunConfigFunc: (ipAddr: string, generatedFileFilepaths: Map<string, string>, staticFileFilepaths: Map<StaticFileID, string>) => [ContainerRunConfig, Error]
+    ): [ServiceContext, Map<string, PortBinding>, Error] { //TODO
+
+        const [serviceContext, hostPortBindings, err] = this.AddServiceToPartition(
+            serviceId,
+            defaultPartitionId,
+            containerCreationConfig,
+            generateRunConfigFunc,
+            )
+        if (err != null) {
+            return [null, null, err ] //TODO no personalized error message
+        }
+
+        return [serviceContext, hostPortBindings, null]
+    }
+
+    // Docs available at https://docs.kurtosistech.com/kurtosis-libs/lib-documentation
+    public AddServiceToPartition(
```

Comment:
remember to make sure all your functions match the JS naming conventions - i.e. starting with lowercase